client/network/src/protocol/generic_proto/behaviour.rs

```diff
@@ -1363,7 +1393,14 @@ impl NetworkBehaviour for GenericProto {
 			}
 		}
 
-		for (peer_id, peer_state) in self.peers.iter_mut() {
+		while let Poll::Ready(Some(peer_id)) = Pin::new(&mut self.peers_check_after).poll_next(cx) {
+			let peer_state = match self.peers.get_mut(&peer_id) {
+				Some(s) => s,
+				// We intentionally never remove elements from `peers_check_after`, and it may
+				// thus contain peers which are now gone. This is a normal situation.
+				None => continue,
+			};
+
 			match peer_state {
 				PeerState::PendingRequest { timer, .. } => {
 					if let Poll::Pending = Pin::new(timer).poll(cx) {
```

Comment:
Related to the other comments, why do we need to keep this `timer` and poll it here?